### PR TITLE
Feature/bake solid anim 104902938

### PR DESCRIPTION
**Pull request by AurL**
Handles solid animation baking and uses the internal API "bake_action" function.
Also adds an option to force baking for all animation, and one to force the use of quaternions

### CODE REVIEW

File: blender-2.5/exporter/osg/osgbake.py [hunk 1]
```diff
@@ -17,261 +17,90 @@
...
+            blender_object.rotation_mode = rotation_mode
+
+        scene.frame_set(current_frame_backup)
+
+    if hasEulerRotationKeyframes(blender_object):
+        generateQuaternionKeyframes(scene, blender_object)
+
+        # remove euler rotation keyframes
+        blender_object.rotation_mode = 'QUATERNION'
+        blender_object.keyframe_delete('rotation_euler')
```
Comment (AurL) [inline blender-2.5/exporter/osg/osgbake.py]: No! Need to keep original keyframes. Duplicate existing, and bake the duplicate

**Comment by cedricpinson** [comment]
what's happen to the scene after the scene is exported. Is there still baked / tmp data in the scene ?

**Comment by AurL** [comment]
I have to change the way action are baked since here the original data is modified. That's the purpose of my comment. 
Btw I was thinking about a way to restore the scene after the export, but it seems difficult to ensure the backup in all cases, so I went back to the current idea which is to generate extra data, export it and then clean it. 
(Note that the data that is not linked to anything in the scene is cleared when saving the scene in .blend )

**Comment by cedricpinson** [comment]
ask to on blender channel on irc or forums. Also we could ask to the guy who helped us to integrate the exporter to blender

**Comment by AurL** [comment]
Ok, I wanted to use the API function "bake_action" but it doesn't handles "use_quaternions" option, so it needs some extra code to actually output quaternions. I think the better way is to actually include the code from this function in osgexport and add the option ourselves (needs something like 3 lines). With this, we don't have to preprocess the data before baking and if we will be able to improve it for other things if we need 
We could also make a submission to include this option in the original code

**Comment by AurL** [comment]
Closing this since skeletal and solid are handled by the same code. Cross referencing with https://github.com/cedricpinson/osgexport/pull/41 to keep track of comments

**Pull request closed without merging**